core/src/main/scala/org/scalatra/ScalatraBase.scala

```diff
@@ -648,31 +662,30 @@ trait ScalatraBase
     removeRoute(HttpMethod(method), route)
   }
 
-  protected[scalatra] def addStatusRoute(codes: Range, action: => Any): Unit = {
+  protected def addStatusRoute(codes: Range, action: => Any): Unit = {
     val route = Route(Seq.empty, () => action, (req: HttpServletRequest) => routeBasePath(req))
     routes.addStatusRoute(codes, route)
   }
 
-  // those methods are there to prevent invoking a macro from code in scalatra-core (which would need separate compile passes)
-  private[scalatra] def before0(transformers: RouteTransformer*)(fun: => Any): Unit = {
+  protected def appendBeforeFilter(transformers: RouteTransformer*)(fun: => Any): Unit = {
     routes.appendBeforeFilter(Route(transformers, () => fun))
   }
 
-  private[scalatra] def after0(transformers: RouteTransformer*)(fun: => Any): Unit = {
+  protected def appendAfterFilter(transformers: RouteTransformer*)(fun: => Any): Unit = {
```

Comment:
It enables a library user to use this method, in the case he does not want or can not use the macro.